binary-search-tree/node-js/inventory-management.js

```diff
@@ -19,15 +23,40 @@ class BinarySearchTree {
     }
 
     search(id) {
-        console.error("you must implement search");
-    }
-
+            return this.searchRecursive(id, this.root);
+        }
+        // console.error("you must implement search");
+        // 1. Start from the root node
+        // 2. Compare the current node with the search value. Return early if found
+        // 3. Determine if we should go left or right
+        // 4. Update the current node based on the decision from (3). Rinse and repeat from (2)
+    searchRecursive(id, currentNode) {
+        if (currentNode === null) {
+            return null ;
+        }
+        else if(currentNode.val.id === id ) {
+            return currentNode.val;
+        }
+        else if (id > currentNode.val.id) {
+            return this.searchRecursive(id, currentNode.right);
+        }
+        else {
+            return this.searchRecursive(id, currentNode.left);
+        }
+    };
+    
     updateInventory(id, inventoryCount) {
-        console.error("you must implement updateInventory");
+        const currentNode = this.searchRecursive(id, this.root);
+        if (currentNode === null) {
+            return;
+        }
+        currentNode.inventoryCount = inventoryCount;
+        return currentNode.inventoryCount;
```

Comment:
The exercise spec doesn't specify a return value for `updateInventory`. Worth reflecting on why you chose to return this value. 

I'm not saying this is wrong. It's just good to be intentional and have a good reason backing a particular design decision.

Reply:
the task is asking us to update `inventoryCount`...so doesn't that mean we need to return a value for `inventoryCount`?

Reply:
No, it doesn't necessarily mean it needs to be return. The task is purely about updating the `inventoryCount`.

Though, if we're talking about being intentional, a good reason for returning the updated inventory count on success and `undefined` otherwise, as your function is currently doing, would be to make it clear to the caller that the operation was successful. So that if I call `bst.updateInventory(6, 20)` my intention is to update the inventory count of the `Product` with `id=6` to `20`. If the product is found and its inventory count is updated then the return value is `20`, otherwise the return value is `undefined`. Without an explicit return value the function would otherwise always return `undefined` and I wouldn't be able to immediately make heads or tales of the outcome.

Reply:
i am explicitly returning value of `inventoryCount`....i'm confused with your explanation....